source/SIL.AppBuilder.Portal.Frontend/src/sockets/data-socket.tsx

```diff
@@ -21,23 +21,50 @@ import { DataClient } from './clients';
 
 import { dataToLocalCache } from '~/data/orbitjs-operations-support/serialize-from-api';
 
+import { useMemoIf } from '~/lib/hooks';
+
+import { DataHub } from './clients/data';
+
+const mockClient: DataClient = {
```

Comment:
ugly hack for tests, :(

Reply:
The library we're using for sockets has no tests as well: https://github.com/sketch7/signalr-client/blob/master/test/test-setup.ts
woo